### PR TITLE
build: enforce conventional commits

### PR DESCRIPTION
**Pull request by diehbria**
## Overview
Enforces basic conventional commit msgs in the CI as well as in local development (so that developers see their commits are not in the correct format when they attempt to create a commit).

The goal of this PR is to improve the developer experience in IoT App Kit.

## Legal
This project is available under the [Apache 2.0 License](http://www.apache.org/licenses/LICENSE-2.0.html).

### CODE REVIEW

**Review by TheEvilDev — DISMISSED** [review]

File: package-lock.json [hunk 1]
```diff
@@ -5805,6 +5808,716 @@
         "node": ">=0.1.90"
       }
     },
+    "node_modules/@commitlint/cli": {
+      "version": "17.4.4",
+      "resolved": "https://registry.npmjs.org/@commitlint/cli/-/cli-17.4.4.tgz",
+      "integrity": "sha512-HwKlD7CPVMVGTAeFZylVNy14Vm5POVY0WxPkZr7EXLC/os0LH/obs6z4HRvJtH/nHCMYBvUBQhGwnufKfTjd5g==",
+      "dev": true,
+      "dependencies": {
+        "@commitlint/format": "^17.4.4",
```
Comment (TheEvilDev) [inline package-lock.json:5817]: Man So many new dependencies.

File: .husky/commit-msg [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env sh
+. "$(dirname -- "$0")/_/husky.sh"
+
+npx --no -- commitlint --edit ${1}
```
Comment (NorbertNader) [inline .husky/commit-msg:4]: Very cool! This assumes we have npx installed though. Let's add this to the documentation or add it to dependencies.

**Review by NorbertNader — CHANGES_REQUESTED** [review]

File: .husky/commit-msg (hunk 2 above)
Comment (diehbria) [reply .husky/commit-msg:4]: npx comes with npm which comes with node so we shouldn't need to document npx specifically

Comment (NorbertNader) [reply .husky/commit-msg:4]: A sweet. It’s bundled with npm since 7.0.0

**Comment by diehbria** [comment]
will address in https://github.com/awslabs/iot-app-kit/pull/671

**Pull request closed without merging**